vdom/src/com/vdom/core/BasePlayer.java

```diff
@@ -4089,7 +4089,13 @@ public boolean settlers_copperIntoHand(MoveContext context, int coppers, int set
     
     @Override
     public HuntingGroundsOption sprawlingCastle_chooseOption(MoveContext context) {
-    	return HuntingGroundsOption.GainDuchy;
+        boolean getEstates = context.game.piles.get(Cards.duchy.getName()).getCount() == 0    //no duchies left
```

Comment:
Part of the idea of getting the three Estates is to have more victory cards to line up better with the Opulent Castle, so I would add a condition to having an Opulent Castle as well.